### PR TITLE
Try stuff out with CodeQL Analysis.

### PR DESCRIPTION
**Pull request by adityaruplaha**
I'm curious to see what static code analysis gives, don't merge this.

### CODE REVIEW

**Comment by adityaruplaha** [comment]
@flavioislima you think this is useful? Vulnerability and performance analysis isn't bad imo.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
```
Comment (flavioislima) [inline .github/workflows/codeql-analysis.yml:36]: Are you sure it will work with Typescript as well?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+  schedule:
+    - cron: '28 19 * * 4'
```
Comment (flavioislima) [inline .github/workflows/codeql-analysis.yml:21]: This means it will run on the main branch whenever it reaches this time?

Comment (adityaruplaha) [reply .github/workflows/codeql-analysis.yml:21]: It should.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (adityaruplaha) [reply .github/workflows/codeql-analysis.yml:36]: I mean, the docs say so.

**Comment by flavioislima** [comment]
I still don't get the point about this QL stuff, I think we could use our time to develop new features and improve the code overall with things more valuable to the user like:
**- Integration with other stores**
That should be the main focus since will bring a lot of value to the product and to the community since we are lacking a really good GOG client on Linux right now.

I think have tests is essential and have an organized codebase is also really good, but whats most important is to have a good features since its what the user sees in the end :P

**Pull request closed without merging**